app/models/stats.rb

```diff
@@ -4,25 +4,34 @@ class Stats
   attr_reader :new_reg, :assisted_pc, :email_renewals
 
   def initialize
-    @new_reg = calculate_new_reg
+    @reg_from_last_week = retrieve_reg_from_last_week
+    @new_reg = count_new_reg
     @assisted_pc = calculate_assisted_pc
     @email_renewals = calculate_email_renewals
   end
 
   private
 
-  def calculate_new_reg
-    from = 7.days.ago.beginning_of_day
-    to = 1.day.ago.beginning_of_day
-
-    WasteExemptionsEngine::Registration.where(submitted_at: from..to, referring_registration_id: nil).count
+  def count_new_reg
+    @reg_from_last_week.where(referring_registration_id: nil).count
   end
 
   def calculate_assisted_pc
-    35.4
+    ((count_assisted_reg.to_f / count_new_reg.to_f) * 100).round(1)
   end
 
   def calculate_email_renewals
     17
   end
+
+  def count_assisted_reg
+    @reg_from_last_week.where.not(assistance_mode: nil).count
```

Comment:
I have a suggestion here then! 
A good practice is to do this: 
```
@_count_assisted_reg ||= @reg_from_last_week.where.not(assistance_mode: nil).count
```
This will ensure the method can still be accessed all the times we like, but the execution of the DB query will only happen once :)